### PR TITLE
add jwt functionality

### PR DESCRIPTION
**Pull request by Mattnite1**

### CODE REVIEW

File: ui/src/app/auth/signup/page.tsx [hunk 1]
```diff
@@ -1,11 +1,103 @@
...
+  const inputValues = {
+    email: "",
+    username: "",
+    password: "",
+    firstName: "",
+    lastName: "",
+  };
+  const [data, setData] = useState(inputValues);
+
+  const handleSubmit = async (e: any) => {
```
Comment (Mattnite1) [inline ui/src/app/auth/signup/page.tsx:26]: zapomnialem usunac tego asynca XD

File: backend/src/auth/auth.service.ts [hunk 2]
```diff
@@ -34,7 +34,7 @@ export class AuthService {
     }
   }
   async login(dto: AuthDto) {
```
Comment (pgorka1) [inline backend/src/auth/auth.service.ts:36]: lacking result type

**Comment by pgorka1** [comment]
Add eslint rule for result required result type of functions https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/explicit-function-return-type.md

File: ui/package.json [hunk 3]
```diff
@@ -9,19 +9,51 @@
...
+    "@radix-ui/react-select": "^2.0.0",
+    "@radix-ui/react-separator": "^1.0.3",
+    "@radix-ui/react-slot": "^1.0.2",
+    "@radix-ui/react-switch": "^1.0.3",
+    "@radix-ui/react-toast": "^1.1.5",
+    "@radix-ui/react-toggle": "^1.0.3",
+    "axios": "^1.6.1",
+    "class-variance-authority": "^0.7.0",
+    "clsx": "^2.0.0",
+    "cors": "^2.8.5",
```
Comment (pgorka1) [inline ui/package.json:31]: remove unused packages, there should be as few of them as possible

File: ui/src/app/api/auth/[...nextauth]/route.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+import GitHubProvider from "next-auth/providers/github"
+import CredentialsProvider from "next-auth/providers/credentials"
+
+export const authOptions = {
+    providers: [
+        GitHubProvider({
+            clientId: process.env.GITHUB_ID ?? "",
+            clientSecret: process.env.GITHUB_SECRET ?? ""
+        }),
+        CredentialsProvider({
```
Comment (pgorka1) [inline ui/src/app/api/auth/[...nextauth]/route.ts:11]: you are not using jwt here, probably needs to be added

File: ui/src/app/api/auth/[...nextauth]/route.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+            name: "Credentials",
+            credentials: {
+                email: {label: "Email", type: "text", placeholder: "email"},
+                password: {label: "Password", type: "password"}
+            },
+            async authorize(credentials, req) {
+                const res = await fetch('http://localhost:3333/auth/login', {
```
Comment (pgorka1) [inline ui/src/app/api/auth/[...nextauth]/route.ts:18]: I would be nice to have server url as variable in .env and concrete endpoints in objects to use it like this fetch(`${ENV_VARIABLE}/${endpoints.auth.login}`)

Comment (pgorka1) [reply ui/src/app/api/auth/[...nextauth]/route.ts:18]: Such usage is less error prone

File: ui/src/app/api/auth/[...nextauth]/route.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+                    method: "POST",
+                    body: JSON.stringify(credentials),
+                    headers: {"Content-Type": "application/json"}
+                });
+                const user = await res.json();
+
+                if (res.ok && user) {
+                    console.log(user.access_token)
```
Comment (pgorka1) [inline ui/src/app/api/auth/[...nextauth]/route.ts:26]: console logs in code are bad practice, there should be eslint rule added for that: https://eslint.org/docs/latest/rules/no-console

File: ui/src/app/api/auth/signup/page.tsx [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+import { FormSchema } from "./zod.signup.schema";
+import { useRouter } from 'next/navigation'
+
+export default function InputForm() {
+  async function handleSubmit(data: z.infer<typeof FormSchema>) {
+    try {
+    const response = await axios.post('http://localhost:3333/auth/signup', data)
+    console.log(response.statusText)
+    if (response.statusText === 'Created') {
+      const router = useRouter()
```
Comment (pgorka1) [inline ui/src/app/api/auth/signup/page.tsx:14]: Using hooks like this not in top level of component may lead to errors: https://legacy.reactjs.org/docs/hooks-rules.html the useRouter needs to be called in InputForm directly

File: ui/src/app/api/auth/signup/zod.signup.schema.tsx [hunk 8]
```diff
@@ -0,0 +1,28 @@
+import path from "path";
```
Comment (pgorka1) [inline ui/src/app/api/auth/signup/zod.signup.schema.tsx:1]: Looks great!

File: ui/src/components/view/CreateAccountView.tsx [hunk 9]
```diff
@@ -0,0 +1,59 @@
+// "use client"
```
Comment (pgorka1) [inline ui/src/components/view/CreateAccountView.tsx:1]: Try not to leave such commented out files

**Pull request merged**